### ISSUE TITLE
Incorrect recognition of enumerated list

### ISSUE THREAD

**Issue by anntzer** [habamax/vim-rst#14]
In
```rst
In this paragraph, I mention a book written by
A. Nonymous, a well-known author.
```
vim-rst currently incorrectly highlights `A.` as if it was an enumerated list entry.

**Comment by habamax** [comment]
This one is tricky.

I can turn off highlighting for `A.` type of list items if there is no other way to fix it.

**Comment by habamax** [comment]
![image](https://user-images.githubusercontent.com/234774/162170206-50521050-4a06-4e5e-b182-5de25efb8971.png)

HTML

![image](https://user-images.githubusercontent.com/234774/162170305-dcf6b2fd-3077-43f1-8c2b-9d78d35a2d10.png)

**Comment by habamax** [comment]
![image](https://user-images.githubusercontent.com/234774/162170727-426d7875-5c92-4926-8edc-16d1a4ad4fd6.png)


HTML

![image](https://user-images.githubusercontent.com/234774/162170795-429204ab-acab-44d7-a3d7-5fd88e43754b.png)

**Comment by habamax** [comment]
![image](https://user-images.githubusercontent.com/234774/162170928-43046223-8fb9-46fc-ad0b-a7b07d158815.png)

HTML

![image](https://user-images.githubusercontent.com/234774/162170962-f8f55491-ad98-4546-8513-694b0a6caa45.png)

**Comment by anntzer** [comment]
Ugh, this is indeed quite tricky.  I think that just checking that the previous line is blank (giving up on the case where a true paragraph starts with `A.`) would still be quite good, though.

**Comment by habamax** [comment]
It is even more complicated :)

![image](https://user-images.githubusercontent.com/234774/162186822-9c1f67ca-9270-4e00-b1a4-25ec8709caf8.png)

HTML

![image](https://user-images.githubusercontent.com/234774/162186877-fb524aed-1f84-4c7f-885f-3647774759ab.png)

**Comment by habamax** [comment]
I would leave it as is -- implementation if any, would really be non trivial.

**Comment by anntzer** [comment]
Fair enough.  I guess I'd rather just not highlight them then; if you want to keep it can you make that configurable?  Thanks :)

**Comment by habamax** [comment]
Could you check the latest master?

**Comment by anntzer** [comment]
It works, thanks for the quick fix!
